GitAutomation/Functions/Send-GitBranch.ps1

```diff
@@ -0,0 +1,100 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# 
+#     http://www.apache.org/licenses/LICENSE-2.0
+# 
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+function Send-GitBranch

```

Comment:
When automating commits to a repo, automation doesn't know what's happening on the remote side of things. It has to be able to handle new commits, etc. These functions are designed for this type of automation first, developer workflows second.

# Send-GitBranch

I'm using the verb `Send` because it is mainly the function's purpose is to push or send changes on a branch to a remote repository. That's network communication. I originally used `Update-GitBranch`, but I don't think `Update` is the right verb.

This function doesn't simply wrap `Send-GitObject` because when automating commits you have to automate pushing those commits. If `Send-GitBranch` didn't exist, every script we have would look like:

    Save-GitChange
    while( -not (Send-Gitobject) )
    {
        Sync-GitBranch
    }

with `Send-GitBranch`, it looks like this:

    Save-GitChange
    Send-GitBranch

`Send-GitBranch` also does a bunch of error handling that most people won't do, which increases its value.

# Sync-GitBranch

The verb Sync is a data verb and not network communication, but it accurately reflects what Git is doing: synchronizing two branches so they are in the same state. In this case, the remote/local branches. Am I sad that Git does this by reading commits from the remote repository? Yes. 

This was originally called `Update-GitBranch`, but I felt that `Update` wasn't descriptive enough. I feel `Sync` is more accurate to what Git is doing.

Reply:
That makes sense as well. I was coming at it from the developer workflow level and wanted to highlight the point of `sync` already being used by other organizations for the workflow in the function we call `Send-GitBranch` and thus the idea of renaming `Send-GitBranch -> Sync-GitBranch` came about, just to keep things consistent.

Also, technically `Send-GitBranch` does also synchronize two branches into the same state, just on a two-way level instead of one-way, but that's just me being nitpicky :stuck_out_tongue_closed_eyes: 

From an automation angle though, this makes sense and works for me. Great discussion, thanks. :+1: